pkg/services/rpcsrv/server.go

```diff
@@ -3075,3 +3077,52 @@ func (s *Server) Addresses() []string {
 	}
 	return res
 }
+
+func (s *Server) getRawNotaryPool(_ params.Params) (any, *neorpc.Error) {
+	if !s.chain.P2PSigExtensionsEnabled() {
+		return nil, neorpc.NewInternalServerError("P2PSignatureExtensions are disabled")
+	}
+	nrp := s.coreServer.GetNotaryPool()
+	res := result.RawNotaryPool{}
+	res.Hashes = make(map[util.Uint256][]util.Uint256)
+	nrp.IterateVerifiedTransactions(func(data any) {
+		d := data.(*payload.P2PNotaryRequest)
+		mainHash := d.MainTransaction.Hash()
+		fallbackHash := d.FallbackTransaction.Hash()
+		res.Hashes[mainHash] = append(res.Hashes[mainHash], fallbackHash)
+	})
+	return res, nil
+}
+
+func (s *Server) getRawNotaryTransaction(reqParams params.Params) (any, *neorpc.Error) {
+	if !s.chain.P2PSigExtensionsEnabled() {
+		return nil, neorpc.NewInternalServerError("P2PSignatureExtensions are disabled")
+	}
+
+	txHash, err := reqParams.Value(0).GetUint256()
+	if err != nil {
+		return nil, neorpc.ErrInvalidParams
+	}
+	nrp := s.coreServer.GetNotaryPool()
+	// try to find fallback transaction
+	tx, ok := nrp.TryGetValue(txHash)
+	if !ok {
+		// try to find main transaction
+		notaryRequest, ok := nrp.TryFindData(func(data any) bool {
+			if data == nil {
+				return false
+			}
+			notaryRequest := data.(*payload.P2PNotaryRequest)
+			return notaryRequest.MainTransaction.Hash() == txHash
+		})
+		if !ok {
+			return nil, neorpc.ErrUnknownTransaction
+		}
+		tx = notaryRequest.(*payload.P2PNotaryRequest).MainTransaction
+	}
+
+	if v, _ := reqParams.Value(1).GetBoolean(); v {
+		return tx, nil
+	}
+	return tx.Bytes(), nil
```

Comment:
@AnnaShaleva could you, please, tell me, is `tx.Bytes()` return value useful? Or parameter `verbose` is needless here too?